### PR TITLE
Add commanline options to the samples code.

### PR DESCRIPTION
**Pull request by azabbasi**
image

### CODE REVIEW

File: sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/DigitalTwinsLifecycleAsyncSample.java [hunk 1]
```diff
@@ -68,23 +64,26 @@
...
+                    .tenantId(parsedArguments.getTenantId())
+                    .clientId(parsedArguments.getClientId())
+                    .clientSecret(parsedArguments.getClientSecret())
                     .build()
             )
-            .endpoint(endpoint)
+            .endpoint(parsedArguments.getDigitalTwinUrl())
             .httpLogOptions(
                 new HttpLogOptions()
                     .setLogLevel(HttpLogDetailLevel.NONE))
```
Comment (abhipsaMisra) [inline sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/DigitalTwinsLifecycleAsyncSample.java:84]: nit: we could get the log level from command line as well

File: sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/SamplesArguments.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    private String clientSecret;
+
+    public SamplesArguments(String[] args) {
+        Options options = new Options();
+        Option input = new Option("d", DIGITALTWINS_URL, false, "DigitalTwins endpoint URL");
+        Option tenantId = new Option("t", TENANT_ID, false, "AAD Tenant Id");
+        Option clientId = new Option("c", CLIENT_ID, false, "AAD Client Id");
+        Option clientSecret = new Option("s", CLIENT_SECRET, false, "AAD Client Secret");
+
+        input.setRequired(true);
```
Comment (abhipsaMisra) [inline sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/SamplesArguments.java:24]: Ah, is there no way to set it inline during initialization? Or some annotation perhaps?

Comment (azabbasi) [reply sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/SamplesArguments.java:24]: unfortunately not :(

File: sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/SamplesArguments.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        tenantId.setRequired(true);
+        clientId.setRequired(true);
+        clientSecret.setRequired(true);
+
+        options.addOption(input);
```
Comment (abhipsaMisra) [inline sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/SamplesArguments.java:29]: nit: Can we do an inline addition to `options`? Just trying to reduce the lines of code here!

**Review by abhipsaMisra — APPROVED** [review]

File: sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/SamplesArguments.java (hunk 3 above)
Comment (azabbasi) [reply sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/SamplesArguments.java:29]: sure, but won't save that much space :))

Comment (azabbasi) [reply sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/SamplesArguments.java:29]:
```
        Options options = new Options()
            .addOption(input)
            .addOption(tenantId)
            .addOption(clientId)
            .addOption(clientSecret);
```

Comment (abhipsaMisra) [reply sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/SamplesArguments.java:29]: How about:
```java
Options options = new Options()
            .addOption(new Option("d", DIGITALTWINS_URL, false, "DigitalTwins endpoint URL"))
            .addOption(new Option("t", TENANT_ID, false, "AAD Tenant Id"))
            .addOption(new Option("c", CLIENT_ID, false, "AAD Client Id"))
            .addOption(new Option("s", CLIENT_SECRET, false, "AAD Client Secret"));
```

There also seems to be a builder options:
```java
new Options()
	.addOption(Option.builder("d")
			.required()
			.hasArg(true)
			.argName(DIGITALTWINS_URL)
			.desc("DigitalTwins endpoint URL")
			.build())
	(...)
	.addOption(Option.builder("t")
			.required()
			.hasArg(true)
			.argName(TENANT_ID)
			.desc("AAD Tenant Id")
			.build());
```

However, I am not too particular about it, just giving you options :)

**Review by bikamani — APPROVED** [review]

File: eng/versioning/external_dependencies.txt [hunk 4]
```diff
@@ -161,6 +161,7 @@ io.opentelemetry:opentelemetry-sdk;0.6.0
 io.grpc:grpc-context;1.30.2
 io.projectreactor:reactor-test;3.3.9.RELEASE
 junit:junit;4.13
+commons-cli:commons-cli;1.3
```
Comment (alzimmermsft) [inline eng/versioning/external_dependencies.txt:164]: Given that this is being used as a test dependency this is fine to add.

**Review by alzimmermsft — APPROVED** [review]

**Pull request merged**